File: backend/models/note.go
package models

import (
	"gorm.io/gorm"
)

type Note struct {
	gorm.Model
	Content   string  `json:"content"`
	ContactID uint    `json:"contact_id"`
	DealID    uint    `json:"deal_id"`
	Contact   Contact `json:"contact" gorm:"foreignKey:ContactID"`
	Deal      Deal    `json:"deal" gorm:"foreignKey:DealID"`
	UserID    uint    `json:"user_id"`
	User      User    `json:"user" gorm:"foreignKey:UserID"`
}

File: frontend/app/src/types/Note.ts
import { User } from './User';
import { Contact } from './Contact';

export interface Note {
  id: number;
  content: string;
  contact_id: number;
  user_id: number;
  created_at: string;
  updated_at: string;
  contact?: Contact;
  user?: User;
}

export interface NoteFormData {
  content: string;
  contact_id: number;
}
